ksqldb-engine/src/main/java/io/confluent/ksql/physicalplanner/PhysicalPlanner.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.physicalplanner;
+
+import io.confluent.ksql.execution.context.QueryContext.Stacker;
+import io.confluent.ksql.execution.plan.ExecutionStep;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.KStreamHolder;
+import io.confluent.ksql.execution.plan.StreamSink;
+import io.confluent.ksql.execution.streams.ExecutionStepFactory;
+import io.confluent.ksql.logicalplanner.LogicalPlan;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.physical.PhysicalPlan;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.serde.FormatInfo;
+import io.confluent.ksql.serde.SerdeFeatures;
+import java.util.Collections;
+import java.util.Optional;
+
+public final class PhysicalPlanner {
```

Comment:
Maybe some Java Docs for the new classes would help explain strategy.  Is it too early to add those?